documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1020 @@
+* [Authorization using the Authorization Code Flow](#authorization-using-the-authorization-code-flow)
+    * [Authentication Request](#authentication-request)
+    * [Authentication Response](#authentication-response)
+        * [Successful Response](#successful-response)
+        * [Error Response](#error-response)
+    * [Access Token Request](#access-token-request)
+    * [Access Token Response](#access-token-response)
+        * [Successful Response](#successful-response-1)
+        * [Error Response](#error-response-1)
+* [Authorization using the OpenID Connect Client-Initiated Backchannel Authentication Flow (CIBA)](#authorization-using-the-openid-connect-client-initiated-backchannel-authentication-flow-ciba)
+    * [Authentication Request](#authentication-request-1)
+    * [Authentication Response](#authentication-response-1)
+        * [Successful Response](#successful-response-2)
+        * [Error Response](#error-response-2)
+    * [Access Token Request](#access-token-request-1)
+    * [Access Token Response](#access-token-response-1)
+        * [Successful Response](#successful-response-3)
+        * [Error Response](#error-response-3)
+* [Client Credentials Flow](#client-credentials-flow)
+    * [Access Token Request](#access-token-request-2)
+    * [Access Token Response](#access-token-response-2)
+        * [Successful Response](#successful-response-4)
+        * [Error Response](#error-response-4)
+* [The Scope Parameter](#the-scope-parameter)
+* [ID Token](#id-token)
+* [Client Authentication](#client-authentication)
+* [References](#references)
+* [Appendix A. RSA Key Used in Examples](#appendix-a-rsa-key-used-in-examples)
+    * [Appendix A.1. Authorization Server RSA Key](#appendix-a1-authorization-server-rsa-key)
+    * [Appendix A.2. Client RSA Key](#appendix-a2-client-rsa-key)
+<!-- TOC -->
+
+
+## Introduction
+
+Telco network capabilities exposed through APIs provide great value to customers. By simplifying the complexity of telco networks with APIs and making the APIs available across telco networks and countries, CAMARA enables easy and seamless access. To ensure seamless interoperability and increased security, this technical document introduces the OpenID Connect (OIDC) profile, which is specifically tailored for CAMARA APIs.
+
+OpenID Connect is a widely adopted standard for user authentication and authorization in the context of APIs. This document builds on the basic principles of OIDC and outlines CAMARA-specific guidelines to address CAMARA APIs scenarios and use cases.
+
+The CAMARA OIDC profile includes three different authorization mechanisms: Authorization Code Flow, Client Initiated Backchannel Authentication (CIBA), and Client Credentials. Each method is detailed, providing comprehensive insight into the authorization and token request processes, mandatory parameters, successful response content or potential error scenarios.
+
+Key areas covered include:
+
+* Authorization flows:
```

Comment:
To define profiles, it is first necessary to establish exactly which standard(s) these profiles apply to. That is not done explicitly in this document (the OpenID CIBA specification is not even referenced). My understanding is that the relevant baseline standards are:
- [OpenID Connect Core 1.0 incorporating errata set 2](https://openid.net/specs/openid-connect-core-1_0.html) for the `authorization_code` grant
- [OpenID Connect Client-Initiated Backchannel Authentication Flow - Core 1.0](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html) for the `urn:openid:params:grant-type:ciba` grant
- [RFC 6749 The OAuth 2.0 Authorization Framework](https://datatracker.ietf.org/doc/html/rfc6749) for the `client_credentials` grant

These standards should be clearly referenced either here or in the individual sections dealing with each grant type. The individual sections should then highlight the differences between the standard specification and the profile, particularly for parameters that are optional in the standard.

Reply:
Direct references to the standards have been added in each of the individual sections of the flows.